packages/connection-form/src/utils/csfle-kms-fields.ts

```diff
@@ -23,13 +23,25 @@ export interface KMSField<KMSProvider extends KMSProviderName> {
   description?: string;
 }
 
+// Best understood in a TS playground: https://tinyurl.com/74pb7cex
+// This type hackery became necessary because some of the KMS options became
+// union types, e.g. { ... } | { accessToken: string }.
+type DecayUnion<T> = {
+  [k in T extends T ? keyof T : never]: (T & Partial<Record<string, never>>)[k];
+} & T;
```

Comment:
Ohh, that's a nice helper for unions with no key overlap! One thing I don't really follow here is this `& T` at the end, doesn't look to me like it's doing anything, but maybe I'm missing something 🤔

Reply:
Mh yeah, it’s not strictly necessary here, but otherwise you can’t assign `DecayUnion<T>` back to `T`, which makes sense to me

(Also, just added a fixup commit because there was a `?` missing here, also not strictly necessary but should be more correct that way)

Reply:
Ahhh, got it! Makes sense